VisualRust/RustCommentSelectionCommandHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace VisualRust
+{
+    class RustCommentSelectionCommandHandler : VSCommandTarget<VSConstants.VSStd2KCmdID>
+    {
+        readonly ITextBuffer Buffer;
+        public RustCommentSelectionCommandHandler(IVsTextView vsTextView, IWpfTextView textView)
+            : base(vsTextView, textView)
+        {
+            Buffer = textView.TextBuffer;
+        }
+
+        protected override bool Execute(VSConstants.VSStd2KCmdID command, uint options, IntPtr pvaIn, IntPtr pvaOut)
+        {
+            var snapshot = Buffer.CurrentSnapshot;
+            int start = TextView.Selection.Start.Position.Position;
+            int end = TextView.Selection.End.Position.Position;
+
+            // this is what we will store start offset in to get maximal indentation amount
+            int? insertStartOffset = null;
+
+            using (var edit = Buffer.CreateEdit())
+            {
+                while (start <= end)
+                {
+                    var line = snapshot.GetLineFromPosition(start);
+                    var text = line.GetText();
```

Comment:
It's only looking at a single line, whereas the article is about getting the text of the *full* buffer snapshot. Thinking about it again, one could also ignore the line when it contains only whitespace, which is what VS does for C# (just checked). In that case I would have to look at the actual text. So what do you prefer?